src/context/directory/handlers/flowVaultConnections.ts

```diff
@@ -0,0 +1,88 @@
+import path from 'path';
+import fs from 'fs-extra';
+import { isEmpty } from 'lodash';
+import { constants } from '../../../tools';
+
+import log from '../../../logger';
+import { getFiles, existsMustBeDir, dumpJSON, loadJSON, sanitize } from '../../../utils';
+import { DirectoryHandler } from '.';
+import DirectoryContext from '..';
+import { ParsedAsset } from '../../../types';
+
+import { FlowVaultConnection } from '../../../tools/auth0/handlers/flowVaultConnections';
+
+type ParsedFlowVaults = ParsedAsset<'flowVaultConnections', FlowVaultConnection[]>;
+
+function parse(context: DirectoryContext): ParsedFlowVaults {
+  const flowVaultsFolder = path.join(context.filePath, constants.FLOWS_VAULT_DIRECTORY);
+  if (!existsMustBeDir(flowVaultsFolder)) return { flowVaultConnections: null }; // Skip
+
+  const files = getFiles(flowVaultsFolder, ['.json']);
+
+  const flowVaultConnections = files.map((f) => {
+    const connection = {
+      ...loadJSON(f, {
+        mappings: context.mappings,
+        disableKeywordReplacement: context.disableKeywordReplacement,
+      }),
+    };
+    return connection;
+  });
+
+  return {
+    flowVaultConnections,
+  };
+}
+
+async function dump(context: DirectoryContext) {
+  const { flowVaultConnections } = context.assets;
+
+  if (!flowVaultConnections || isEmpty(flowVaultConnections)) return; // Skip, nothing to dump
+
+  // Check if there is any duplicate form name
+  const vaultConnectionsNames = flowVaultConnections.map((form) => form.name);
+  const duplicateVaultConnectionsNames = vaultConnectionsNames.filter(
+    (name, index) => vaultConnectionsNames.indexOf(name) !== index
+  );
+
```

Comment:
@kushalshit27 Just a suggestion.

Multiple loops are used to identify the duplicates.

If you look at the code
- Line 43, has the time complexity of O(N)
- Line 44, 45, 46 collectively has the complexity of O(N2), this can be improved to O(N) and the whole operation can be performed in a single pass.
- Line 50 -> O(N), this can be improved to O(1)
Line -> 55 O(N), this can be improved to O(1)



This improvement can be extended across other files as well (`flows.ts`, `forms.ts` etc).